spec-tests/src/test/java/org/candlepin/spec/consumers/ConsumerResourceSpecTest.java

```diff
@@ -107,6 +110,64 @@ public void shouldPopulateGeneratedFieldsWhenCreatingConsumers() {
             .isBeforeOrEqualTo(post);
     }
 
+    public static Stream<Arguments> cloudFactsSource() {
+        List<String> awsFacts = List.of("aws_instance_id", "aws_account_id", "aws_billing_products",
+            "aws_marketplace_product_codes");
+
+        List<String> azureFacts = List.of("azure_instance_id", "azure_subscription_id", "azure_offer");
+
+        List<String> gcpFacts = List.of("gcp_instance_id", "gcp_project_id", "gcp_license_codes");
+
+        return Stream.of(
+            Arguments.of(awsFacts),
+            Arguments.of(azureFacts),
+            Arguments.of(gcpFacts));
+    }
+
+    @ParameterizedTest
+    @MethodSource("cloudFactsSource")
+    public void shouldAllowRegistrationWithNullDataInCloudFacts(List<String> facts) throws Exception {
+        ApiClient adminClient = ApiClients.admin();
```

Comment:
We can remove this line or use this adminClient instead of the adminClient that is created in the "setup" method. This also includes `shouldAllowRegistrationWithEmptyDataInCloudFacts`.

Reply:
Fixed